Packs/CommonScripts/Scripts/AnalyzeTimestampIntervals/AnalyzeTimestampIntervals.py

```diff
@@ -0,0 +1,141 @@
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+import statistics
+
+
+def calculate_interval_differences(timestamps):
+    if len(timestamps) < 2:
+        demisto.debug("Not enough timestamps to calculate intervals.")
+        return []
+
+    # Sort the timestamps
+    timestamps.sort()
+
+    # Calculate the time differences between each consecutive pair (in seconds)
+    intervals = [(timestamps[i + 1] - timestamps[i]) / 1000 for i in range(len(timestamps) - 1)]  # Convert to seconds
+
+    demisto.debug(f"Calculated intervals: {intervals}")
+    return intervals
+
+def check_high_frequency(timestamps, max_intervals_per_window, time_window=60):
+    """ Check if there is a high number of intervals within a short time window (in seconds). """
+    timestamps.sort()
+    count_exceeds_threshold = False
+
+    for i in range(len(timestamps)):
+        window_start = timestamps[i]
+        window_end = window_start + (time_window * 1000)  # time_window in seconds, converting to milliseconds
+
+        count = sum(1 for t in timestamps if window_start <= t <= window_end)
+
+        if count > max_intervals_per_window:
+            count_exceeds_threshold = True
+            break
+
+    return count_exceeds_threshold
+
+def calculate_statistics(intervals):
+    if not intervals:
+        demisto.debug("No intervals to calculate statistics.")
+        return None, None, None
+
+    mean_interval = sum(intervals) / len(intervals)
+    median_interval = statistics.median(intervals)
+    std_deviation = statistics.stdev(intervals) if len(intervals) > 1 else 0
+
+    return mean_interval, median_interval, std_deviation
+
+def analyze_intervals(timestamps, verbose, max_intervals_per_window=30, interval_consistency_threshold=0.15):
```

Comment:
Taking a look. The reason for verbose being False by default is because the expectation is that these lists can be huge (over 500 sometimes) which would flood the war-room since it returns all of the intervals in seconds.